src/Adapter/Order/CommandHandler/DeleteProductFromOrderHandler.php

```diff
@@ -70,6 +100,49 @@ public function handle(DeleteProductFromOrderCommand $command)
         $order = $order->refreshShippingCost();
 
         Hook::exec('actionOrderEdited', ['order' => $order]);
+
+        $this->contextStateManager->restoreContext();
+    }
+
+    /**
+     * @param Order $order
+     * @param OrderDetail $orderDetail
+     * @param Cart $cart
+     *
+     * @return bool
+     */
+    private function updateCart(Order $order, OrderDetail $orderDetail, Cart &$cart)
+    {
+        $oldCartRules = $cart->getCartRules();
+        // Update Product Quantity in the cart
+        $return = $cart->updateQty(
+            $orderDetail->product_quantity,
+            $orderDetail->product_id,
+            $orderDetail->product_attribute_id,
+            false,
+            'down'
+        );
+        // Remove & Add CartRules applied to cart
+        CartRule::autoRemoveFromCart();
+        CartRule::autoAddToCart();
+
+        // Remove previous cart rules applied to order
+        $result = array_diff(
+            array_map('serialize', $oldCartRules),
+            array_map('serialize', $cart->getCartRules())
+        );
+        $result = array_map('unserialize', $result);
+
+        foreach ($order->getCartRules() as $orderCartRule) {
```

Comment:
This is technically not updating the cart, but I get that you need to do it here because you need the diff of cart rules
Maybe rename the method then `updateCartAndRules` ?

Reply:
I agree with @jolelievre , the name of the method misled my understanding when reading this code for the first time

Reply:
Fixed ;)